cpg-language-cxx/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/ExpressionHandler.kt

```diff
@@ -429,7 +429,12 @@ class ExpressionHandler(lang: CXXLanguageFrontend) :
         when {
             reference is MemberExpression -> {
                 val baseType = reference.base.type.root
-                assert(baseType !is SecondOrderType)
+                if (baseType is SecondOrderType) {
```

Comment:
I think you can remove these two lines alltogether, because `baseType` is not used anymore.